test/minikube/minikube_kaa_additions_test.go

```diff
@@ -118,4 +120,55 @@ func TestKaaRolebindingDisabled(t *testing.T) {
 	require.True(t, len(config.StatefulSets) == 0)
 	require.True(t, len(config.Volumes) == 0)
 	require.True(t, len(config.Pods) == 0)
-}
\ No newline at end of file
+}
+
+func TestKubernetesTopologyDiscover(t *testing.T) {
+	testlib.SkipTestOnNuoDBVersionCondition(t, "< 6.0.3")
+	testlib.AwaitTillerUp(t)
+	defer testlib.VerifyTeardown(t)
+
+	currentRegions := testlib.LabelNodesIfMissing(t, "topology.kubernetes.io/region", "test-region")
+	currentZones := testlib.LabelNodesIfMissing(t, "topology.kubernetes.io/zone", "test-zone")
+	currentNodes := testlib.GetNodeNames(t)
+
+	defer testlib.Teardown(testlib.TEARDOWN_ADMIN)
+
+	options := helm.Options{}
+	helmChartReleaseName, namespaceName := testlib.StartAdmin(t, &options, 1, "")
+	kubectlOptions := k8s.NewKubectlOptions("", "", namespaceName)
+	options.KubectlOptions = kubectlOptions
+	admin := fmt.Sprintf("%s-nuodb-cluster0", helmChartReleaseName)
+	admin0 := fmt.Sprintf("%s-0", admin)
+
+	testlib.VerifyLabels(t, namespaceName, admin0,
+		map[string][]string{
+			"node":   currentNodes,
+			"zone":   currentZones,
+			"region": currentRegions,
```

Comment:
I'm not sure why these are _plural_ because one AP can be running in a single region, zone, and node. I guess this is working because in _minikube_ there is only one node.

Reply:
There will be multiple values in the off-chance that someone tries to run the test in a multi-region cluster.

Reply:
You could add a test (in another PR) that runs against the AWS multi-cluster and exercises the admin <-> engine affinity and labels on a multi-region/multi-node cluster.